### ISSUE TITLE
Idea - add command on 2.x to read 1.x config file and generate 2.x version

### ISSUE THREAD

**Issue by SpacePossum** [FriendsOfPHP/PHP-CS-Fixer#2309]
We could warn the user were needed and/or prompt with question for tweaking.
What do you think?

**Comment by keradus** [comment]
they are not compatible interfaces, classes are in different namespaces, and basically to generate... new file? I'm not fan of generating PHP code that much ;P

**Comment by SpacePossum** [comment]
>I'm not fan of generating PHP code that much ;P

can't tell if you are serious : P

I'll write it anyway for my own projects, so I can PR it so you can see if you like it.
If not it just ends up in my monstrous-unofficial-all-rejected-PRs-Fixer repo ^_^

**Comment by keradus** [comment]
generating file... no... but using new Fixer to replace method calls/imports.... actually, why not :D

**Comment by SpacePossum** [comment]
I'll see what I can come up with, closing till than :)

**Comment by keradus** [comment]
i guess it will be problematic to update ruleset.

**Comment by glensc** [comment]
i would had liked some kind of helper to map rules to new names

even if it's some sed script:

```
sed -e 's/php_closing_tag/no_closing_tag/'
```

@SpacePossum did you came up with something? share please!

**Comment by SpacePossum** [comment]
I took a diff. approach where I was polyfilling all classes and interfaces from the 1.x line needed to read a configuration file for the 1.x line in the 2.x version, than translating the rules and writing back in the new config format. Given the amount of code involved (including a new command), time needed to stabilize, writing docs and such and not so much interest for it by others I abandoned the idea, sorry man not much of value I can share, never made it beyond a rough POC.
